framework/test/gtest/BundleActivatorTest.cpp

```diff
@@ -27,39 +27,67 @@
 #include "cppmicroservices/FrameworkEvent.h"
 #include "cppmicroservices/FrameworkFactory.h"
 
+#include "TestUtilBundleListener.h"
 #include "TestUtils.h"
-#include "TestingConfig.h"
-#include "TestingMacros.h"
+
+#include "gtest/gtest.h"
 
 using namespace cppmicroservices;
 
+class BundleActivatorTest : public ::testing::Test
+{
+protected:
+  static void SetUpTestSuite()
+  {
+    f = std::make_shared<cppmicroservices::Framework>(
+      FrameworkFactory().NewFramework());
+    f->Start();
+
+    EXPECT_NO_THROW(ctx = std::make_shared<cppmicroservices::BundleContext>(
+                      f->GetBundleContext()));
+  }
+
+  static void TearDownTestSuite()
+  {
+    f->Stop();
+    f->WaitForStop(std::chrono::milliseconds::zero());
+    f = nullptr;
+    ctx = nullptr;
+  }
+
+  static std::shared_ptr<cppmicroservices::Framework> f;
+  static std::shared_ptr<cppmicroservices::BundleContext> ctx;
```

Comment:
I don't believe these should be static. Its undesired as it re-uses the state of the Framework across multiple tests. I know this is how the old tests were structured however I think that is fundamentally incorrect. Tests shouldn't rely on side-effects from other tests, unless that's the explicit point of the test.